samples/matter/light_bulb/sample.yaml

```diff
@@ -4,41 +4,85 @@ sample:
 tests:
   sample.matter.light_bulb.debug:
     build_only: true
+    extra_args: DCONFIG_CHIP_LIB_SHELL=y
     integration_platforms:
       - nrf52840dk_nrf52840
       - nrf5340dk_nrf5340_cpuapp
-    platform_allow: nrf52840dk_nrf52840 nrf5340dk_nrf5340_cpuapp
+    platform_allow: nrf52840dk_nrf52840 nrf5340dk_nrf5340_cpuapp nrf21540dk_nrf52840
```

Comment:
nrf21540dk_nrf52840 doesn't have external flash and hence cannot be used in variants that have DFU enabled (all except `no_dfu`).